src/Name/Generator/RequestNameGenerator.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Jaeger\Symfony\Name\Generator;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\Event\TerminateEvent;
+
+class RequestNameGenerator implements NameGeneratorInterface, EventSubscriberInterface
+{
+    private array $generators;
+
+    private string $route;
+
+    public function __construct(array $generators = [])
+    {
+        $this->generators = $generators;
+    }
+
+    public function add(string $regexp, NameGeneratorInterface $generator): RequestNameGenerator
+    {
+        $this->generators[sprintf('/%s/', $regexp)] = $generator;
+
+        return $this;
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [RequestEvent::class => ['onRequest', 30], TerminateEvent::class => ['onTerminate', -16384],];
+    }
+
+    public function onRequest(RequestEvent $event): void
+    {
+        $this->route = $event->getRequest()->attributes->get('_route', $event->getRequest()->getRequestUri());
```

Comment:
Same comment as for `ControllerNameGenerator`

```suggestion
        $this->route = (string)$event->getRequest()->attributes->get('_route', $event->getRequest()->getRequestUri());
```

Reply:
Still actual